util/tlgen/validate.py

```diff
@@ -282,6 +287,13 @@ def validate(obj: OrderedDict) -> Xbar:  # OrderedDict -> Xbar
 
                 addr_entry = (address_from, address_to)
 
+                if checkBaseSizeOverlap(address_from, size):
+                    log.error(
+                        "Size mask and base address are overlapping. "
+                        " Check the config. Addr(0x%x - 0x%x)"
+                        % (addr_entry[0], addr_entry[1]))
+                    raise SystemExit("Base/size overlapping error occurred")
```

Comment:
@weicaiyang added this check to avoid running into this address decoding issue again.